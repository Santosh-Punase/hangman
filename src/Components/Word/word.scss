@import 'src/assets/scss/variables';

.hangman__word {
  position: absolute;
  display: flex;
  justify-content: space-between;
  bottom: 2rem;
  left: 2rem;
  right: 2rem;

  &__char-box {
    width: 6vw;
    height: 6vw;
    border-radius: 0.3rem;
    background-color: $gray-dark;
    font-size: 2.25rem;
    font-weight: 800;
    font-style: bold;
    color: $white;
    text-transform: uppercase;
    text-align: center;
    line-height: 6vw;

    &.box-empty {
      background-color: $gray-light;
    }
  }

}