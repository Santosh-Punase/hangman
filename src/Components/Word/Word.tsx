import React from 'react';

import './word.scss';

interface Props {
  word: string;
  gussedCharacters: string[];
}

const Word: React.FC<Props> = ({ word, gussedCharacters }) => {
  if(!word) {
    return null;
  }

  const startIndex = 11 - word.length;

  return (
    <div className="hangman__word">
      {[...Array(11)].map((x,i) => (
        startIndex - i > 0
        ? <div className="hangman__word__char-box box-empty" key={i}></div>
        : <div className="hangman__word__char-box" key={i}>{gussedCharacters.includes(word.charAt(i-startIndex)) ? word.charAt(i-startIndex) : ''}</div>
      ))
      }
    </div>
  );
}

export { Word };
