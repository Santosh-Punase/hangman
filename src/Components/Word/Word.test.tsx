import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { Word } from './Word';

afterEach(cleanup);

describe('Guess Word', () => {

  test('should not render initially', () => {
    const { asFragment } = render(<Word word='' gussedCharacters={['a','b']} />);
    
    expect(asFragment()).toMatchInlineSnapshot(`<DocumentFragment />`);
  });

  test('should render guessed characters', () => {
    const { asFragment } = render(<Word word='analog' gussedCharacters={['a','b']}  />);
    
    expect(asFragment()).toMatchSnapshot();
  });  
});
