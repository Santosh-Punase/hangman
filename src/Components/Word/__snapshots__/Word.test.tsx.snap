// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guess Word should render guessed characters 1`] = `
<DocumentFragment>
  <div
    class="hangman__word"
  >
    <div
      class="hangman__word__char-box box-empty"
    />
    <div
      class="hangman__word__char-box box-empty"
    />
    <div
      class="hangman__word__char-box box-empty"
    />
    <div
      class="hangman__word__char-box box-empty"
    />
    <div
      class="hangman__word__char-box box-empty"
    />
    <div
      class="hangman__word__char-box"
    >
      a
    </div>
    <div
      class="hangman__word__char-box"
    />
    <div
      class="hangman__word__char-box"
    >
      a
    </div>
    <div
      class="hangman__word__char-box"
    />
    <div
      class="hangman__word__char-box"
    />
    <div
      class="hangman__word__char-box"
    />
  </div>
</DocumentFragment>
`;
